docs/adr/0024-upgrade-templating-toolset-of-auth-server-emails.md

```diff
@@ -0,0 +1,237 @@
+# Upgrade the templating toolset of Auth Server emails
+
+- Status: proposed
+- Deciders: Yogita Bhatia, Lauren Zugai, Jody Heavener, Vijay Budhram, [TBD]
+- Date: 2021-05-24
+
+Technical Story: [#8600](https://github.com/mozilla/fxa/issues/8600)
+
+## Context and Problem Statement
+
+[Mustache](https://www.npmjs.com/package/mustache), used in `fxa-auth-server` email templates, can accommodate very limited logic in its templates which has been a pain point for Firefox Accounts engineers and discussion around using a different templating system [began here](https://github.com/mozilla/fxa/issues/4627). While converting our emails to a more modern templating solution, there is an opportunity to evaluate what stack would be the most ideal for FxA emails beyond a proposed templating solution. This includes evaluating our CSS options, improving how we can preview emails in various states, and our localization tools, and the best approach for landing new templates in production.
+
+## Decisions and Considered Options
+
+### Templating
+
+- Option A - Continue to use Mustache
+- Option B - Use React server-side to generate static pages
+- Option C - Use EJS
+
+### CSS styles
+
+- Option A - Continue using ad-hoc inline styles
+- Option B - Use TailwindCSS
+- Option C - Use CSS-in-JS
+- Option D - Use a CSS pre-processor like SASS
+
+### Email previewing
+
+- Option A - Continue to use the `write-emails` command and maildev
+- Option B - Use Storybook
+- Option C - Any other client that could be used to view emails? [todo]
```

Comment:
Yeah I looked into a few alternatives like [Nodemailer](https://www.npmjs.com/package/nodemailer), [Mailtrap](https://mailtrap.io/) and [MailHog](https://github.com/mailhog/MailHog) but couldn't really find anything different that could benefit us. Although, Mailtrap (paid service) could be an interesting choice to test the mails since it is an online tool, the QA team won't have to touch the codebase.

Reply:
Cool, I'm going to mark this as resolved since you added Mailtrap as a considered option 😁